Stratis.Bitcoin/Features/RPC/Models/BlockHeaderModel.cs

```diff
@@ -0,0 +1,84 @@
+﻿using NBitcoin;
+using NBitcoin.DataEncoders;
+
+namespace Stratis.Bitcoin.Features.RPC.Models
+{
+#pragma warning disable IDE1006 // Naming Styles (ignore lowercase)
+
+    /// <summary>
+    /// Data structure for RPC block headers
+    /// 
+    /// <see cref="https://bitcoin.org/en/developer-reference#getblockheader"/>
+    /// </summary>
+    public class BlockHeaderModel
+    {
+        /// <summary>
+        /// Constructs a RPC BlockHeaderModel from a block header object
+        /// </summary>
+        /// <param name="blockHeader">the block header</param>
+        public BlockHeaderModel(BlockHeader blockHeader)
+        {           
+            if (blockHeader != null)
+            {
+                this.version = (uint)blockHeader.Version;
+                this.previousblockhash = blockHeader.HashPrevBlock.ToString();
+                this.merkleroot = blockHeader.HashMerkleRoot.ToString();
+                this.time = blockHeader.Time;
+                byte[] bytes = this.GetBytes(blockHeader.Bits.ToCompact());
+                string encodedBytes = Encoders.Hex.EncodeData(bytes);
```

Comment:
Are you sure this is according to the bitcoin spec? I'm not sure myself, not a tricky question :-)
https://github.com/bitcoin/bitcoin/blob/041dad94b047313a17edb0324742cf80cfd550f1/src/rpc/blockchain.cpp#L650
https://github.com/bitcoin/bitcoin/blob/041dad94b047313a17edb0324742cf80cfd550f1/src/rpc/blockchain.cpp#L94